src/types/products.ts

```diff
@@ -133,12 +133,19 @@ interface RootDocsPath {
 	addOverviewItem?: boolean
 }
 
+export type DocsNavItem = {
+	icon: string
+	label: string
+	fullPath: string
+}
+
 interface ProductData extends Product {
 	algoliaConfig: {
 		indexName: string
 	}
 	basePaths: string[]
 	rootDocsPaths: RootDocsPath[]
+	docsNavItems?: (DocsNavItem | string)[]
```

Comment:
To me, it feels clearer to say that `DocsNavItem` can be a `string` or the object, rather than saying `docsNavItems` can be an array of strings or an array of the objects.

```ts
export type DocsNavItem = string | {
	icon: string
	label: string
	fullPath: string
}

interface ProductData extends Product {
	algoliaConfig: {
		indexName: string
	}
	basePaths: string[]
	rootDocsPaths: RootDocsPath[]
	docsNavItems?: DocsNavItem[]
}
```

Reply:
This makes sense...  as-is though, a `DocsNavItem` is the specific `{ icon, label, fullPath }` structure, and when authoring `docsNavItems`, authors have the option to use a `string` to reference a `rootDocsPath.path` that will then get parsed into a `docsNavItem`. This change alone would break things elsewhere, so have deferred it for now. Added a comment to clarify that - definitely happy to revisit later, agree it could be more clear & clean 👍